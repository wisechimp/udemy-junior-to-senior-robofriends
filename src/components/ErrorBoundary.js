import React, { Component } from 'react'

class ErrorBoundary extends Component {
  constructor(props) {
    super(props)
    this.state = {
      hasError: false
    }
  }

  // This is like the try-catch block in js
  componentDidCatch(error, info) {
    this.setState({ hasError: true })
  }

  render() {
    if (this.state.hasError) {
      return <h1 id='errorMessage'>Oops, there's been an error.</h1>
    }
    return this.props.children
  }
}

export default ErrorBoundary