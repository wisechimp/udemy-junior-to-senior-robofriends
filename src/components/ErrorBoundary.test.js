import { shallow } from 'enzyme'
import React from 'react'
import ErrorBoundary from './ErrorBoundary'

describe('ErrorBoundary Component Tests', () => {
  const wrapper = shallow(<ErrorBoundary />)

  it('expect to render ErrorBoundary component', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('recognises that an error has occured', () => {
    wrapper.setState({ hasError: true })
    expect(wrapper.state().hasError).toBeTruthy()
    expect(wrapper.find('[id="errorMessage"]').length).toBe(1)
  })

  it('performs as expected when there is no error', () => {
    wrapper.setState({ hasError: false })
    expect(wrapper.state().hasError).toBeFalsy()
    expect(wrapper.find('[id="errorMessage"]').length).toBe(0)
  })
})
