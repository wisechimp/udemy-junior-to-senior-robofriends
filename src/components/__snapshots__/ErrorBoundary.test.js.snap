// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary Component Tests expect to render ErrorBoundary component 1`] = `ShallowWrapper {}`;
